CRM/Logging/ReportDetail.php

```diff
@@ -218,25 +218,57 @@ protected function convertDiffsToRows() {
           $to = implode(', ', array_filter($tos));
         }
 
+        $cfArray = [];
         $tableDAOClass = CRM_Core_DAO_AllCoreTables::getClassForTable($table);
         if (!empty($tableDAOClass)) {
           $tableDAOFields = (new $tableDAOClass())->fields();
           // If this field is a foreign key, then we can later use the foreign
           // class to translate the id into something more useful for display.
           $fkClassName = $tableDAOFields[$field]['FKClassName'] ?? NULL;
+        } 
+        else {
+          /* Since this table didn't match a core table, check if it's a custom field. */
+          $cfResultSet = \Civi\Api4\CustomField::get()
+            ->addSelect('id', 'data_type', 'fk_entity', 'option_group_id')
+            ->addWhere('column_name', '=', $field)
+            ->setLimit(1)
+            ->execute();
+          if(is_array($cfResultSet->first())){
+            $cfArray = $cfResultSet->first();
+          }
         }
         if (isset($values[$field][$from])) {
           $from = $values[$field][$from];
         }
         elseif (!empty($from) && !empty($fkClassName)) {
           $from = $this->convertForeignKeyValuesToLabels($fkClassName, $field, $from);
         }
+        elseif (!empty($from) && is_numeric($from) && array_key_exists("id", $cfArray) && is_numeric($cfArray["id"])) {
+          /* Translate the id into something more useful for display, namely for id's that refer to option values and contacts. */
+          $fromAsArray = civicrm_api3('CustomValue', 'getdisplayvalue', [
+            'entity_id' => $this->cid,
+            'custom_field_id' => $cfArray["id"],
+            'custom_field_value' => $from,
+          ]);
+          $from = $this->formatLabelAndIdForDisplay($fromAsArray['values'][$cfArray["id"]]['display'],$from);
```

Comment:
There are a couple situations where this gives an error because the typehint on $keyval is int but $from can be numeric but not an int (money fields, float fields, or even a text field where the value happens to be a number). Two possible suggestions:

1. Since the formatting as "label: value" doesn't make sense in those situations and seeing the internal value is kind of optional even when it does make sense, don't bother calling formatLabelAndIdForDisplay and just leave it as the display label as returned from the api.
2. Instead of is_numeric, check that it's an integer, since while it might be possible for a lookup value to be a float, that's not likely if hitting this else block. e.g. for contact reference it's going to be an int.

Reply:
Thanks, I'll fix it and test a bit more.

Reply:
I made a commit following both of your suggestions, after testing with other non-integer numeric custom field types. Commit statements contain more details... but first...

1. Regarding point 1, I display the label returned from the api, unless that label comes back as blank. In that case $from or $to don't get changed.
2. Regarding point 2, since the $from and $to are always strings, I chose instead to verify that the custom field `data_type` is an integer, thus narrowing the formatting scope to custom fields of types: Int, ContactReference, and EntityReference (all of which would be integers).

Reply:
I also added a tiny one-line bug fix for legacy code that was affecting my custom field formatting code. Let me know if you prefer that I pull this out as a separate PR.

**BEFORE**
![image](https://github.com/civicrm/civicrm-core/assets/112041049/569c2787-9093-4ae2-a0c0-c04abce5e2d6)

![image](https://github.com/civicrm/civicrm-core/assets/112041049/702bc2a6-e3cb-489a-825d-f7f5b149d98f)

**AFTER**
![image](https://github.com/civicrm/civicrm-core/assets/112041049/178dc534-956d-49c2-95a3-a75757a7fb70)